controllers/profiles.js

```diff
@@ -0,0 +1,118 @@
+const fs = require("fs")
+const profiles = require("../data/profiles.json")
+
+const errorResHelper = (errorDescription, res, errCode = 400) => {
+  res.status(errCode).json({ error: errorDescription })
+}
+
+const createProfile = (req, res) => {
+  const now2ISO = new Date().toISOString()
+  const missing = []
+  const newProfile = {
+    idProfile: new Date().getTime().toString(),
+    name: req.query.name ?? missing.push("'name'"),
+    created: {
+      admin: req.query.idAdmin ?? missing.push("'idAdmin'"),
+      timestamp: now2ISO,
+    },
+    modified: {},
+  }
+
+  if (missing.length) {
+    return errorResHelper(
+      `queryParam ${missing.join(" and ")} is missing.`,
+      res
+    )
+  }
+
+  if (profiles.filter((el) => el.name === req.query.name).length > 0) {
+    return errorResHelper(
+      `Profile name '${req.query.name}' already exists.`,
+      res
+    )
+  }
+
+  profiles.push(newProfile)
+
+  fs.writeFile("./data/profiles.json", JSON.stringify(profiles), (err) => {
+    if (err) {
+      errorResHelper(err, res)
+    } else {
+      res.status(201).json(newProfile)
+    }
```

Comment:
if you return each `res.status(...` you don't need to use an else condition.

```
if (err) {
  return errorResHelper(err, res)
}
return res.status(201).json(newProfile)
```

Reply:
apply this logic to the entire file

Reply:
The `return` in the last `res.status...`  seems redundant. Is it ok to omit it?